app/create.js

```diff
@@ -0,0 +1,90 @@
+/* file: form.js
+ * authors: H. Bouakaz, S. de Vliet, S. de Jong & E. Werkema
+ * date: 19/3/2016
+ * version 1.1
+ *
+ * Description:
+ */
+
+define(['app/database', 'jquery', 'bootstrap', 'parsley', 'app/selectLanguage', 'app/selectSubject', 'app/date'], function (db, $, bootstrap, parsley, language, subject, date) {
+	var numberOfFormItems = 0;
+	var formItemId = 0;
+
+	// Function for removing elements from the form
+	function remove_element(element) {
+		if (numberOfFormItems > 1) {
+			element.parents("tr").remove();
+			numberOfFormItems--;
+		}
+	}
+
+	// Function for adding elements to the form
+	function add_element() {
+		var newElement = $('#item-layout').clone(true).appendTo("#items table").removeAttr("id");
+		var add_id = newElement.html().replace(/{i}/g, formItemId).replace(/{required}/g, 'required=""');
+		newElement.html(add_id);
+		newElement.on("click", ".remove", function() {
+			remove_element($(this));
+		});
+		numberOfFormItems++;
+		formItemId++;
+	}
+	// Add the first element
+	add_element();
+
+	$(".add").click(function(add) {
+		add_element();
+		return false;
+	});
+
+	$('#items input[type="text"]:last').on('keydown', function(e) {
+	  if (e.which == 9) { // If a tab is pressed
+	      add_element();
+	  }
+	});
+
+	// Check in the database if the name of the dataset already exists
+	window.Parsley.addValidator('datasetName', {
+		validateString: function(value, requirement) {
+			var result = db.getQuery("getDatasetByName", [value]);
```

Comment:
I agree with this; in as far that multiple datasets can share the same name. However, for the user, it may be confusing if two datasets within the same module share the same name. So I would propose having unique names per module. I made a card for this here: https://trello.com/c/08rDmAhR